src/highdicom/legacy/sop.py

```diff
@@ -25,384 +36,2630 @@
+            if k1 not in ds2:
+                return False
+            elem2 = ds2[k1]
+            if not cls.isequal(elem2.value, elem1.value):
+                return False
+        return True
+
+    @classmethod
+    def tag2str(cls, tg: BaseTag) -> str:
+        """Converts tag to hex format like (group, element)
+
+        """
+
+        if not isinstance(tg, BaseTag):
+            tg = Tag(tg)
+        return '(0x{:0>4x}, 0x{:0>4x})'.format(tg.group, tg.element)
+
+    @classmethod
+    def tag2kwstr(cls, tg: BaseTag) -> str:
+        """Converts tag to keyword and (group, element) form
+
+        """
+
+        return '{}-{:32.32s}'.format(
+            cls.tag2str(tg), keyword_for_tag(tg))
+
+
+class GeometryOfSlice:
+
+    """A class for checking dicom slices geometry/parallelity"""
+
+    def __init__(self,
+                 row_vector: ndarray,
+                 col_vector: ndarray,
+                 top_left_corner_pos: ndarray,
+                 voxel_spacing: ndarray,
+                 dimensions: tuple) -> None:
+        """
+        Parameters
+        ----------
+        row_vector: numpy.ndarray
+            3D vector representing row of the input slice
+        col_vector: numpy.ndarray
+            3D vector representing column the input slice
+        top_left_corner_pos: numpy.ndarray
+            3D point representing top left corner position of the input slice
+        voxel_spacing: numpy.ndarray
+            Three element array. 1st and 2nd copied from PixelSpacing and the
+            3rd copied from SliceThickness
+        dimensions: tuple
+            3 element tuple holding x as number of cols, y as number of rows
+            and z as 1
+        """
+
+        self.row_vector = row_vector
+        self.col_vector = col_vector
+        self.top_left_corner_position = top_left_corner_pos
+        self.voxel_spacing = voxel_spacing
+        self.Dim = dimensions
+
+    def get_normal_vector(self) -> ndarray:
+        """Returns the normal vector of the input slice
+
+        """
+
+        n: ndarray = cross(self.row_vector, self.col_vector)
+        n[2] = -n[2]
+        return n
+
+    def get_distance_along_origin(self) -> float:
+        """Returns the shortest distance of the slice from the origin
+
+        """
+
+        n = self.get_normal_vector()
+        return float(
+            dot(self.top_left_corner_position, n))
+
+    def are_parallel(
+            slice1: Any,
+            slice2: Any,
+            tolerance: float = 0.0001,
+        ) -> bool:
+        """Returns False if two slices are not parallel else True
+
+        """
+
+        if (not isinstance(slice1, GeometryOfSlice) or
+                not isinstance(slice2, GeometryOfSlice)):
+            logger.warning(
+                'slice1 and slice2 are not of the same '
+                'type: type(slice1) = {} and type(slice2) = {}'.format(
+                    type(slice1), type(slice2)))
+            return False
```

Comment:
Summarising various comments about this method here in this suggestion. Mypy does not complain about this
```suggestion
    @staticmethod
    def are_parallel(
            slice1: GeometryOfSlice,
            slice2: GeometryOfSlice,
            tolerance: float = 0.0001,
        ) -> bool:
        """Returns False if two slices are not parallel else True."""
        if (not isinstance(slice1, GeometryOfSlice) or
                not isinstance(slice2, GeometryOfSlice)):
            raise TypeError("Inputs slices must be of type 'GeometryOfSlice'")
```